Classes/Service/DemandService.php

```diff
@@ -322,6 +325,77 @@ protected function getPropertiesForDemandedTables(array $tables, array $demands)
             }
         }
 
-        return array_merge_recursive($demandedProperties, $demands);
+        $filteredDemands = $this->filterDemandedProperties($demandedProperties, $demands);
+
+        return array_merge_recursive($demandedProperties, $filteredDemands);
+    }
+
+    /**
+     * Add sorting to QueryBuilder from demands.
+     *
+     * @param QueryBuilder $queryBuilder
+     * @param array $table Array of tables where key used as alias and value used as tablename.
+     * @return QueryBuilder
+     */
+    public function getSortBy(QueryBuilder $queryBuilder, array $table): QueryBuilder
```

Comment:
Use the same order for the arguments as in getRestrictions(). That's best-practice for an API.

```suggestion
     * @param array $table Array of tables where key used as alias and value used as tablename.
     * @param QueryBuilder $queryBuilder
     * @return QueryBuilder
     */
    public function getSortBy(array $table, QueryBuilder $queryBuilder): QueryBuilder
```